tests/test_cases/forkandopen.py

```diff
@@ -5,15 +5,10 @@
 
 def forkandopen(results):
     fdlist = re.findall('[-\d]+', results)
-    print fdlist
-    assert(int(fdlist[-1]) != -1)
     assert(int(fdlist[-4]) != -1)
```

Comment:
What is this doing. Would be good to leave a comment on what this test achieves.

Reply:
I've added an explanation on an earlier commit but it seems to be overwritten by the latest commit.

The function checks the PID/fd values of both lind and regular outputs and makes sure the Parent PID is successful (not returning -1), which also means a child process has been created.

_Detailed:_
A sample fdlist would look like this: ['371320', '371320', '-1', '0', '0', '-1'], with the order of appearance.

Parent: Testing open in parent: pid 371320!
Parent: Pid 371320 opened fd -1
Child: Testing open in child: pid 0!
Child: Pid 0 opened fd **-1**

assert(int(fdlist[-1]) != -1) checks if the child PID's return -1. This would mean failure in the parent process, but not in child processes. Which is why I've removed it.
assert(int(fdlist[-4]) != -1)  makes sure that the parent PID won't return -1.

I'll add the explanation on the py test as well.

Reply:
I think fdlist is a confusing word as the things in the list are not fds